core/src/com/unciv/models/ruleset/Building.kt

```diff
@@ -255,9 +256,45 @@ class Building : RulesetStatsObject(), INonPerpetualConstruction {
                 UniqueType.Unbuildable ->
                     yield(RejectionReasonType.Unbuildable.toInstance())
 
-                UniqueType.OnlyAvailable ->
-                    if (!unique.conditionalsApply(civ, cityConstructions.city))
+                UniqueType.OnlyAvailable -> {
+                    var displayable = false
+                    var shouldYield = false
+                    for (conditional in unique.conditionals) {
+                        if (Conditionals.conditionalApplies(unique, conditional, StateForConditionals(civ, cityConstructions.city)))
+                            continue
+                        when (conditional.type) {
+                            UniqueType.ConditionalBuildingBuiltAmount -> {
+                                val building = civ.getEquivalentBuilding(conditional.params[0]).name
+                                val amount = conditional.params[1].toInt()
+                                val cityFilter = conditional.params[2]
+                                val numberOfCities = civ.cities.count {
+                                    it.cityConstructions.containsBuildingOrEquivalent(building) && it.matchesFilter(cityFilter)
+                                }
+                                if (numberOfCities < amount)
+                                {
+                                    displayable = true
+                                    yield(RejectionReasonType.RequiresBuildingInSomeCities.toInstance(
+                                        "Requires a [$building] in at least [$amount] cities" +
+                                            " ($numberOfCities/$numberOfCities)"))
+                                }
+                            }
+                            UniqueType.ConditionalBuildingBuiltAll -> {
+                                val building = civ.getEquivalentBuilding(conditional.params[0]).name
+                                val cityFilter = conditional.params[1]
+                                if(civ.cities.any { it.matchesFilter(cityFilter)
+                                    !it.isPuppet && !it.cityConstructions.containsBuildingOrEquivalent(building)
```

Comment:
I'm missing an && between the matchesFilter and the !it.isPuppet - what am I missing here?

Reply:
That's... Huh... No warning. I guess kotlin assumes a semicolon here. Weird

The `!it.isPuppet` is now redundant anyways... But huh